### ISSUE TITLE
Integration screen shows 2 devices under mobile_app

### ISSUE THREAD

**Issue by dshokouhi** [home-assistant/home-assistant-android#65]
1. Install latest firebase test app (22 or 23)
2. Login to Home Assistant (latest version 0.102.1)
3. Navigate to integration screen
4. Select mobile app

There are 2 devices for my phone listed.  One of which has not much data and the other shows the `device_tracker` entity (pending #8) as well as some blank automations.

expected: only 1 device should be shown

![image](https://user-images.githubusercontent.com/1634145/69503255-9f7b5a00-0ecc-11ea-8a3d-4783f27b55ff.png)

![image](https://user-images.githubusercontent.com/1634145/69503260-a4d8a480-0ecc-11ea-9307-67257d70bf20.png)

![image](https://user-images.githubusercontent.com/1634145/69503261-aace8580-0ecc-11ea-8763-8705bd3f9012.png)

**Comment by balloob** [comment]
I _think_ that this is a bug in the mobile app integration in Home Assistant.

**Comment by andriej** [comment]
OT: but how to install firebase version? I'd like to help with the bugreports

**Comment by dshokouhi** [comment]
This is the entry I see in `core.device_registry`

```
{
                "area_id": null,
                "config_entries": [
                    "config_entry_id"
                ],
                "connections": [],
                "id": "connection_id_1",
                "identifiers": [
                    [
                        "device_id",
                        "identifier_1"
                    ],
                    [
                        "webhook_id",
                        "identifier_2"
                    ]
                ],
                "manufacturer": "Google",
                "model": "Pixel 3 XL",
                "name": "Pixel 3 XL",
                "name_by_user": null,
                "sw_version": "29",
                "via_device_id": null
            },
            {
                "area_id": null,
                "config_entries": [
                    "config_entry_id"
                ],
                "connections": [],
                "id": "connection_id_2",
                "identifiers": [
                    [
                        "mobile_app",
                        "identifier_1"
                    ]
                ],
                "manufacturer": "Google",
                "model": "Pixel 3 XL",
                "name": null,
                "name_by_user": null,
                "sw_version": "29",
                "via_device_id": null
            }
```

So everything looks similar except one entry is missing the `webhook_id` and the `name` for the device shows as `null`

**Comment by andriej** [comment]
Same on 22, two entries, one empty, one with entity

**Comment by CedrickFlocon** [comment]
@JBassett have you experiment the same issue ?

**Comment by balloob** [comment]
That's weird, devices should be linked based on overlapping identifiers. Definitely a bug in mobile_app/Home Assistant.

**Comment by JBassett** [comment]
@CedrickFlocon Now that someone pointed it out yes.  I thought it was from me wiping the app data and testing.

**Comment by towerhand** [comment]
Same on iOS.

**Comment by olbjan** [comment]
I tested this fix on 0.103.0b0, ran through onboarding on the iOS app again and ended up with the same issue.
